dataprocspawner/spawner.py

```diff
@@ -929,7 +931,32 @@ def _build_cluster_config(self, cluster_data=None):
         uri_geo_slice=-3,
         expected_geo=self.region
       )
-   
+    
+    for server_group in ['master_config', 'worker_config', 'secondary_worker_config']:
+      if server_group in cluster_data['config']:
+        # We do not check the zone because the user form overwrites it.
+        # if 'zone_uri' in cluster_data['config']['gce_cluster_config']:
+        #   self._check_uri_geo(
+        #     uri=cluster_data['config']['gce_cluster_config']['zone_uri'],
+        #     uri_geo_slice=-1,
+        #     expected_geo=self.zone
+        #   )
+        # Machine types must be in the same zone as the Dataproc Cluster.
+        if 'machine_type_uri' in cluster_data['config'][server_group]:
```

Comment:
Could we strip the zone from these entirely? That seems friendlier than forcing the config to match the spawned zone.

Reply:
I thought of that actually. 
For example, replacing (https://www.googleapis.com/compute/v1/)projects/[projectId]/zones/us-east1-a/machineTypes/n1-standard-2 with n1-standar-2 every time?

Reply:
Yep, and the same thing with accelerators.

Reply:
Done in https://github.com/GoogleCloudDataproc/jupyterhub-dataprocspawner/pull/14/commits/47dd837a10e7ef8d1abe001a81be76fef56dae87